comprl/server/__main__.py

```diff
@@ -64,6 +64,13 @@ def on_timeout(self, player: IPlayer, failure, timeout):
         log.debug(f"Player {player.id} had timeout after {timeout}s")
         player.disconnect(reason=f"Timeout after {timeout}s")
 
+    def on_remote_error(self, player: IPlayer, error: Exception):
+        """gets called when there is an error in deferred"""
+        if player.is_connected:
+            log.error(f"Connected player caused remote error \n {error.with_traceback}")
```

Comment:
Maybe we can consider swapping `error.with_traceback` to just `error` as then the complete error and file/method it occurred is printed and not only a cryptical short version. As when an error with a logged-in player happens, this is interesting and might not be expected behavior.

I.e. when raising `raise ValueError("This is a test error")` in `_update()` in the game logic we get with  `error.with_traceback` just:
```ERROR:root:Connected player caused remote error 
 <built-in method with_traceback of Failure object at 0x00000273AA5260B0>
```

 But with  just `error`:
```
ERROR:root:Connected player caused remote error 
 [Failure instance: Traceback: <class 'ValueError'>: This is a test error
c:\Users\pacos\Documents\GitHub\teamproject-competition-server\.conda\Lib\site-packages\twisted\protocols\amp.py:1029:ampBoxReceived
c:\Users\pacos\Documents\GitHub\teamproject-competition-server\.conda\Lib\site-packages\twisted\protocols\amp.py:957:_answerReceived
c:\Users\pacos\Documents\GitHub\teamproject-competition-server\.conda\Lib\site-packages\twisted\internet\defer.py:874:callback     
c:\Users\pacos\Documents\GitHub\teamproject-competition-server\.conda\Lib\site-packages\twisted\internet\defer.py:981:_startRunCallbacks
--- <exception caught here> ---
c:\Users\pacos\Documents\GitHub\teamproject-competition-server\.conda\Lib\site-packages\twisted\internet\defer.py:1075:_runCallbacks
C:\Users\pacos\Documents\GitHub\teamproject-competition-server\comprl\server\networking.py:214:<lambda>
C:\Users\pacos\Documents\GitHub\teamproject-competition-server\comprl\server\interfaces.py:162:_res
C:\Users\pacos\Documents\GitHub\teamproject-competition-server\examples\hockey\hockey_game.py:64:_update
]
```
which is in my opinion more useful

Reply:
Thanks, for the suggestion. You are right, that makes more sense.